far/stddlg.cpp

```diff
@@ -202,17 +186,17 @@ int GetSearchReplaceString(
 		ReplaceDlg[dlg_button_selection].Flags |= DIF_HIDDEN;
 	}
 
-	if (!pCaseSensitive)
+	if (!Options.CaseSensitive.has_value())
```

Comment:
Yes, I know, `std::optional::operator!` does exactly the same thing. However, with `std::optional<bool>` omitting `has_value()` may be confusing to a human reader. I will remove it if you like the terser notation.

Reply:
This is fine, it can be confusing indeed.